loadgen/loadgen.cc

```diff
@@ -180,7 +183,7 @@ struct ResponseDelegateDetailed : public ResponseDelegate {
     // For some reason, using std::unique_ptr<std::vector> wasn't moving
     // into the lambda; even with C++14.
     std::vector<uint8_t>* sample_data_copy = nullptr;
-    if (mode == TestMode::AccuracyOnly) {
+    if (mode == TestMode::AccuracyOnly || acc_log_dist(acc_log_rng) <= acc_log_prob ) {
```

Comment:
Instead of generating the random number here, I think it would be better to add a flag to SampleMetadata call log_response_data or something like that.

That way:
1) which samples are logged do not depend on the response order.
2) we don't need to put a mutex lock around the acc_log_rng. (this might be called by multiple threads at the same time.)
3) we avoid measuring random number generation performance.

Reply:
Thanks for your feedback Brian.
I share your concerns and have thought about a solution like what you propose, but I didn't want to add a flag to the sample metadata--I was concerned that a nefarious SUT could read that flag and choose to only perform real inferences for flagged samples.
I don't think RNG performance itself would be an issue, but I didn't think about needing a mutex.
I had wrote a micro to measure the speed of acc_log_dist(acc_log_rng) calls and got over 6M/s.

Reply:
That makes sense and 6M/s should be plenty fast. To avoid a lock, maybe we could have a thread-local rng owned by the TlsLogger? (There are no other locks in the response code path atm.)

Reply:
Btw, are you able to run the mlperf_loadgen_perftests?
If so, are the results consistent enough for you to tell how much the rng affects performance?

Reply:
I haven't tried mlperf_loadgen_perftests
Are there instructions on how to run them?

Reply:
I need to add some documentation for how to run the tests.

It's only supported with the gn/ninja approach atm:
`third_party/ninja/ninja -C out/Release mlperf_loadgen_perftests && out/Release/mlperf_loadgen_perftests
`

Also, I notice the tests/BUILD.gn file is broken. (An "offline.cc" reference needs to be removed.)